src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs

```diff
@@ -350,6 +360,115 @@ private async Task<(DelegationCheckResponse Result, ServiceResource Resource, Pa
             return (result, resource, fromParty);
         }
 
+        private async Task<(DelegationActionResult Result, ServiceResource Resource, Party FromParty, List<AttributeMatch> To)> ValidateRevoke(DelegationActionType delegationAction, DelegationLookup delegation, int authenticatedUserId)
+        {
+            DelegationActionResult result = new DelegationActionResult { From = delegation.From, To = delegation.To, Rights = new List<RightDelegationResult>() };
+
+            if (delegation.Rights.Count == 0)
+            {
+                result.Errors.Add("Rights", "Request must contain at least one right to be delegated");
+                return (result, null, null, null);
+            }
+
+            // Verify request is for single resource, app or Altinn 2 service
+            if (delegation.Rights.Count > 1 && !ValidateAllRightsAreForTheSameResource(delegation.Rights))
+            {
+                result.Errors.Add("Rights", "Rights delegation only support requests where all rights are for the same resource, app or Altinn 2 service");
+                return (result, null, null, null);
+            }
+
+            DelegationHelper.TryGetResourceFromAttributeMatch(delegation.Rights[0].Resource, out ResourceAttributeMatchType _, out string resourceRegistryId, out string org, out string app, out string serviceCode, out string serviceEditionCode);
+
+            ServiceResource resource = await _contextRetrievalService.GetResourceFromResourceList(resourceRegistryId, org, app, serviceCode, serviceEditionCode);
+            if (resource == null || !resource.Delegable)
```

Comment:
Unsure if it makes sense to block revoke of rights if the resource is changed to no longer be Delegable.
Existing delegated rights should still be revoke-able.

Reply:
aaah, good catch!